services/oauth/oauth.go

```diff
@@ -129,6 +148,51 @@ func (authErrHandler *OAuthErrResHandler) Setup() {
 	authErrHandler.destAuthInfoMap = make(map[string]*AuthResponse)
 	authErrHandler.refreshActiveMap = make(map[string]bool)
 	authErrHandler.disableDestActiveMap = make(map[string]bool)
+	authErrHandler.RudderFlowType = RudderFlow_Delivery
+}
+
+func GetAuthType(config map[string]interface{}) AuthType {
+	var lookupErr error
+	var authValue interface{}
+	if authValue, lookupErr = misc.NestedMapLookup(config, "auth", "type"); lookupErr != nil {
+		return ""
+	}
+	authType, ok := authValue.(string)
+	if !ok {
+		return ""
+	}
+	return AuthType(authType)
+}
+
+// This function creates a new OauthErrorResponseHandler
+func NewOAuthErrorHandler(provider tokenProvider, options ...func(*OAuthErrResHandler)) *OAuthErrResHandler {
+	oAuthErrResHandler := &OAuthErrResHandler{TokenProvider: provider}
+	oAuthErrResHandler.setup()
```

Comment:
we can get rid of the `setup` method
```suggestion
	oAuthErrResHandler := &OAuthErrResHandler{
		tokenProvider:    provider,
		logger:           pkgLogger,
		tr:               &http.Transport{},
		client:           &http.Client{Timeout: config.GetDuration("HttpClient.oauth.timeout", 30, time.Second)},
		destLockMap:      make(map[string]*sync.RWMutex),
		accountLockMap:   make(map[string]*sync.RWMutex),
		lockMapWMutex:    &sync.RWMutex{},
		destAuthInfoMap:  make(map[string]*AuthResponse),
		refreshActiveMap: make(map[string]bool),
		rudderFlowType:   RudderFlow_Delivery,
	}
```

Reply:
Done